src/resolve/resolved-etc-hosts.c

```diff
@@ -444,5 +444,5 @@ int manager_etc_hosts_lookup(Manager *m, DnsQuestion* q, DnsAnswer **answer) {
                         return r;
         }
 
-        return 1;
+        return dns_answer_isempty(*answer) ? 0 : 1;
```

Comment:
That rationale makes sense to me API-wise. But we need that check, as without it this entirely breaks family specific lookups (systemd-resolve -4 or -6). So you liked the previous version where it puts the `dns_answer_isempty()` check into `dns_query_try_etc_hosts()` (the caller) better?